hail/src/main/scala/is/hail/types/physical/PPrimitive.scala

```diff
@@ -82,4 +82,6 @@ class PPrimitiveCode(val pt: PType, val code: Code[_]) extends PCode {
   def memoize(cb: EmitCodeBuilder, name: String): PValue = defaultMemoizeImpl(cb, name)
 
   def memoizeField(cb: EmitCodeBuilder, name: String): PValue = defaultMemoizeFieldImpl(cb, name)
+
+  def primCode[T]: Code[T] = coerce[T](code)
```

Comment:
I assume you're not using `PCode.tcode` because that will go away but this will stay? Probably a good idea to copy the implementation of `tcode`, which checks that the `T` agrees with the ptype.

Reply:
correct. I'll add that.